examples/nanoChatGPT/train_rlhf.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+
+import torch
+
+from data.shakespeare import get_dataloaders
+from env import RLHFEnv
+from models.actor_critic import init_actor_critic
+from models.reward import init_reward_model
+from shared import setup
+from tensordict.nn import set_skip_existing
+
+from torchrl.objectives import ClipPPOLoss
+from torchrl.objectives.value import GAE
+from utils import load_and_update_config
+
+HERE = Path(__file__).parent
+
+
+def main():
+    config = load_and_update_config("config/train_rlhf.yaml")
+    setup(config)
+
+    # ######## INIT MODELS ########
+    actor, critic = init_actor_critic(config)
+
+    reward_model, _ = init_reward_model(config)
+
+    # ######## INIT TRAINING FUNCTIONS ########
+    # Advantage
+    adv_fn = GAE(value_network=critic, gamma=0.99, lmbda=0.95, average_gae=True)
+    # FIXME: why not using the scheduler?
+    # Loss
+    loss_fn = ClipPPOLoss(actor, critic, gamma=0.99)
+
+    # Optimizer
+    optimizer = torch.optim.AdamW(loss_fn.parameters(), lr=1e-3)
+
+    # DataLoader
+    train_loader, _ = get_dataloaders(config)
+
+    # Environment
+    env = RLHFEnv(reward_model=reward_model, config=config, dataloader=train_loader)
+
+    # ######## TRAINING LOOP ########
+
+    def get_action(td):
+        critic(td)
+        actor(td)
+        td["sample_log_prob"] = td["sample_log_prob"].detach()
+        return td
+
+    for i in range(config["max_iters"]):
+        td = env.rollout(
+            config["episode_length"], policy=get_action, return_contiguous=False
+        )
+
+        # TODO: add replay buffer?
+        with set_skip_existing(True):
+            adv_fn(td)
+            loss_vals = loss_fn(td)
```

Comment:
putting loss under skip_existing is dangerous. 
The best is to pass the actor and *just* the critic head to the ppo loss (not the full critic).

Reply:
See for instance `get_value_head` here https://pytorch.org/rl/reference/generated/torchrl.modules.tensordict_module.ActorCriticOperator.html#torchrl.modules.tensordict_module.ActorCriticOperator